pilot/pkg/serviceregistry/federation/controller.go

```diff
@@ -994,7 +1007,7 @@ func (c *Controller) watch(eventCh chan *federationmodel.WatchEvent, stopCh <-ch
 	}
 	req.Header.Add("discovery-service", c.discoveryServiceName)
 	req.Header.Add("remote", fmt.Sprint(common.RemoteChecksum(c.remote)))
-	resp, err := http.DefaultClient.Do(req)
+	resp, err := http.DefaultClient.Do(req.WithContext(ctx))
```

Comment:
does this ensure we'll stop the http client when the context is cancelled?

Reply:
Yes. When the context is cancelled, the blocking read is aborted and the decoder returns the `io.EOF` error, which is handled below.

Reply:
perfect, thanks